src/realm-settings/user-profile/AttributesGroupForm.tsx

```diff
@@ -0,0 +1,263 @@
+import type { UserProfileGroup } from "@keycloak/keycloak-admin-client/lib/defs/userProfileConfig";
+import {
+  ActionGroup,
+  ActionList,
+  ActionListItem,
+  Button,
+  Flex,
+  FlexItem,
+  FormGroup,
+  PageSection,
+  Text,
+  TextContent,
+  TextInput,
+} from "@patternfly/react-core";
+import { MinusCircleIcon, PlusCircleIcon } from "@patternfly/react-icons";
+import React, { useEffect, useMemo } from "react";
+import { SubmitHandler, useFieldArray, useForm } from "react-hook-form";
+import { useTranslation } from "react-i18next";
+import { Link, useHistory, useParams } from "react-router-dom";
+import { FormAccess } from "../../components/form-access/FormAccess";
+import { ViewHeader } from "../../components/view-header/ViewHeader";
+import { useRealm } from "../../context/realm-context/RealmContext";
+import "../RealmSettingsSection.css";
+import type { EditAttributesGroupParams } from "../routes/EditAttributesGroup";
+import { toUserProfile } from "../routes/UserProfile";
+import { useUserProfile } from "./UserProfileContext";
+
+enum AnnotationType {
+  String = "string",
+  Unknown = "unknown",
+}
+
+type StringAnnotation = {
+  type: AnnotationType.String;
+  key: string;
+  value: string;
+};
+
+type UnknownAnnotation = {
+  type: AnnotationType.Unknown;
+  key: string;
+  value: unknown;
+};
+
+type Annotation = StringAnnotation | UnknownAnnotation;
+
+function parseAnnotations(input: Record<string, unknown>) {
+  return Object.entries(input).map<Annotation>(([key, value]) => {
+    if (typeof value === "string") {
+      return { type: AnnotationType.String, key, value };
+    }
+
+    return { type: AnnotationType.Unknown, key, value };
+  });
+}
+
+function transformAnnotations(input: Annotation[]): Record<string, unknown> {
+  return Object.fromEntries(
+    input
+      .filter((annotation) => annotation.key.length > 0)
+      .map((annotation) => [annotation.key, annotation.value] as const)
+  );
+}
+
+type FormFields = Required<Omit<UserProfileGroup, "annotations">> & {
+  annotations: Annotation[];
+};
+
+const defaultValues: FormFields = {
+  annotations: [{ type: AnnotationType.String, key: "", value: "" }],
+  displayDescription: "",
+  displayHeader: "",
+  name: "",
+};
+
+export default function AttributesGroupForm() {
+  const { t } = useTranslation();
+  const { realm } = useRealm();
+  const { config, save } = useUserProfile();
+  const history = useHistory();
+  const params = useParams<Partial<EditAttributesGroupParams>>();
+  const form = useForm<FormFields>({ defaultValues });
```

Comment:
Why not use a destructure here:
```suggestion
  const { control, register, errors, reset } = useForm<FormFields>({ defaultValues });
```

Reply:
okay, I can see why you feel this way, atm there are 50 components that already do it like that

Reply:
Yeah, that's unfortunate.